Framework/DataHandling/inc/MantidDataHandling/MaskDetectors.h

```diff
@@ -80,14 +80,28 @@ class DLLExport MaskDetectors : public API::Algorithm {
   void init() override;
   void exec() override;
   void execPeaks(DataObjects::PeaksWorkspace_sptr WS);
-  void fillIndexListFromSpectra(std::vector<size_t> &indexList,
-                                const std::vector<specnum_t> &spectraList,
-                                const API::MatrixWorkspace_sptr WS);
-  void appendToIndexListFromWS(std::vector<size_t> &indexList,
-                               const API::MatrixWorkspace_sptr maskedWorkspace);
+  void
+  fillIndexListFromSpectra(std::vector<size_t> &indexList,
+                           const std::vector<specnum_t> &spectraList,
+                           const API::MatrixWorkspace_sptr WS,
+                           const std::tuple<size_t, size_t, bool> &range_info);
```

Comment:
Structs are part of the core language as well and I think the use cases for tuples in C++ lay elsewhere. But as you say, this is partly a question of taste and if you feel the improvement in code readability does not justify the amount of changes, then I think we should leave the tuple as it is.